### DISCUSSION TITLE
were these changes supposed to happen? (direct_zoom)

### DISCUSSION THREAD

**Discussion by Tigertye** [CorsixTH/CorsixTH#2218] [Research Room]
If this is the wrong place for this post I apologize.
I finally got tired of all the crashes in my ver 60 and decided, yesterday, to install the ver 66. After playing for about 4 hours I was really frustrated over several changes that were not listed on the releases page. While I am seriously considering going back to ver 60 I thought maybe I should find out if these changes were made deliberately first. I can't decipher half of what you talk about on this site so I decided to just ask.

** when you hit 'pause' the word 'pause' no longer shows in the bar at the bottom? - that makes it kind of hard to tell the difference between a crash and a pause at a glance.
** the radiators have been severely limited as to where they can and cannot be placed? no longer able to place them as sharing the spot of the soda machine access or the sofa access. can no longer place them too close to any machine or door for the same reasons.
** the bins & fire extinguishers footprint is changed and can no longer occupy the same grid square? 
** the sofas can no longer be placed as mirror images because the access spots can no longer overlap? sofas can no longer be placed immediately next to a door, at a right angle because the access spots cannot overlap?
** in build mode the blueprint of the walls on the southish sides have changed somehow. it's now almost impossible to see the difference in where i have actually placed a door or window and where i have not. it's like it blends into the darker blue of the floor part of the blueprint now.

I've not actually had a chance to play ver 66 yet as I am still building the base rooms for my hospital. The fact that every room layout I have ever found to be successful in the past now has to be reworked is truly frustrating. Were these things intended to be changed or were they just accidents? Perhaps some are deliberate and some aren't? I am almost lost with ver 66.

Thanks for any help or explanations you can give,
Tigertye

**Comment by lewri** [comment]
>** when you hit 'pause' the word 'pause' no longer shows in the bar at the bottom? - that makes it kind of hard to tell the difference between a crash and a pause at a glance.

This sounds like a regression potentially. However, it appears that the change means "PAUSE" doesn't come up if you enable "build while paused" in the settings. This one *might* be my fault, so yay for that...

The others would need more time to confirm.

**Reply by lewri** [reply]
Cause is here:
https://github.com/CorsixTH/CorsixTH/commit/41ceb0fe70ee8c40f6663acc6ae811410f22ea01
Previously there was a bug affecting the bottom panel if "build while paused" was on. Probably needs some tweaking to make the pause go away when you're using the panel while paused.

**Reply by Tigertye** [reply]
Interesting. I didn't understand most of that but I think I got the gist of it. I always use "build while paused" so I don't have to worry about things blowing up when I'm not looking. I never realized how important (to me) having it say "PAUSE" in the bottom panel was until it didn't do it any more. It sure made being called away from my computer easier to deal with. One glance and you always know when it's paused.

**Reply by lewri** [reply]
To note: Pause text now reimplemented but allows dynamic text to override temporarily.

**Comment by mugmuggy** [comment]
Radiators. bins, fire extinguishers and door changes all bring alignment to behaviour of Theme Hospital. Obviously it removes the simplicity in some room layouts, when you pile everything into a corner. Bins are now also corridor items. so that can help alleviate the litter concern.

The door space for sofas as an example also came hand in hand with the multiple objects issue. As it was possible to build rooms with no paths. So we ensure the doors have clear ways, all objects are removed from rooms when removed, and multiple object confusion, doesn't break navigation.

As for the blueprint on the walls, I can't recall a change, looks the same as the original game though.

**Reply by Tigertye** [reply]
OK that kind of makes sense. I can live without trying to find space for bins inside the rooms since  they seem to not affect gameplay at all. Bins in the corridors have been there since v60 that I know of and they are VERY useful with the trash situation. I love my bins. So I guess I just need to replan where to place all my radiators now. I can do this.

A room with no path? I wonder if by cramming every room as tightly full as possible if I was causing my own crashes in v60.

The blueprint walls is a huge change in color shades. I have the orig game, v60 and v66 all installed on my puter and there is a distinct difference from 66 to the others. I believe v60 and the orig are using the same shades of color. I have no idea about any other versions. 

Thanks for explaining the path thing.

**Reply by Tigertye** [reply]
I opened all my versions of the game this morning just to be sure I wasn't going blind and yes there is a distinct difference in the build mode blueprints.

Sorry about the size of the pics. The top one is from the orig game, the center is from corsix v60 and the bottom from corsix v66. I knew I was having a hard time seeing those window and door lines. Someone changed the colors a bit and it's not for the better.

The reason I am persuing this color thing is anyone else (like me) who has color vision issues, even slight ones, might not be able to see the window placements at all.
![TH orig game build mode](https://user-images.githubusercontent.com/42595518/185994767-ca06ba71-7bf8-4085-943c-75855ef6ff5f.jpg)
![corsix v60 build mode](https://user-images.githubusercontent.com/42595518/185994778-4976e076-657c-4aff-93e3-d8d1f57738a6.jpg)
![corsix v66 build mode](https://user-images.githubusercontent.com/42595518/185994784-3db06298-e3a9-4728-afc3-a40ff8020904.jpg)

**Comment by Alberth289346** [comment]
Never saw the circle and arrows in the center, so that's definitely missing in Corsix :)

As for the bottom picture, does it get better when you reset the zoom?  
The rectangle with the cursor is not equally large in the bottom 2 screens while the bottom panel does seem equally wide.  
If you have a non-integer zoom, pixels get scaled and blend between pixels, that might give less sharp colors.

**Reply by Tigertye** [reply]
I never noticed that the centering marks were missing :)

Zoom in or out it doesn't change any of the colors or clarity.
They probably were not identical zooms as I just hit the plus sign until it was big enough to show what I was talking about.
I have no idea what a non-integer zoom is. I have whatever is built into the game.

**Comment by mugmuggy** [comment]
Direct zoom is on by default in 0.66. You can turn that off in the config file, albeit with potential performance impacts. That restores the old zoom and renders it correctly. Unfortunately the thin line widths with the zoom dissolve into the background with it on.

**Reply by Tigertye** [reply]
Direct zoom... Is that different than just hitting the plus key?
Or are you saying thats why the windows and doors are so hard to see in blueprint mode?

**Reply by mugmuggy** [reply]
It is the reason why its so hard to see.  https://github.com/CorsixTH/CorsixTH/wiki/Configuration-File look for direct_zoom and set to false instead of nil.

**Reply by Tigertye** [reply]
ooo 
I have never seen this file before. Fascinating. I found the direct zoom line and changed it to false. THANKYOU so much for that! 
You mentioned potential performance impact, what should I look for in case this happens?

I notice this also: 
check_for_updates = true
what do I change that to so it will stop trying? I don't allow internet with the OS I have corsix installed under.

**Reply by mugmuggy** [reply]
false

**Reply by lewri** [reply]
> It is the reason why its so hard to see. https://github.com/CorsixTH/CorsixTH/wiki/Configuration-File look for direct_zoom and set to false instead of nil.

@flackr any ideas why direct zoom might've made blueprints less optimal?

**Reply by Tigertye** [reply]
thankyou :)

**Reply by flackr** [reply]
> > It is the reason why its so hard to see. https://github.com/CorsixTH/CorsixTH/wiki/Configuration-File look for direct_zoom and set to false instead of nil.
> 
> @flackr any ideas why direct zoom might've made blueprints less optimal?

This is because scaling transparent textures results in very poor quality of thin lines (they become partially transparent combining with their adjacent transparent pixels). Without direct_zoom we render the full frame and then scale the lines with the background of what they're drawn on top of. This is similar to #1966, though unlike #1966 there are other ways we could fix it without drawing a full-sized intermediate buffer. e.g. we could render a scaled version of those tiles where non-transparent pixels are preserved / preferred to transparent pixels, or alternately we could draw an intermediate buffer for each floor tile location as long as we ensured that the blueprint was drawn in the correct pass for it.

**Comment by ARGAMX** [comment]
`direct_zoom` `true` (or nil)

<img width="636" height="680" alt="image111" src="https://github.com/user-attachments/assets/4323ba18-604b-4a0f-b18d-3af225f28be4" />

.
.
`direct_zoom` `false`:

<img width="721" height="685" alt="image222" src="https://github.com/user-attachments/assets/5042dc11-f8de-4dd0-8a23-98c2905feb48" />
